### PR TITLE
Visually improve time-series chrome component

### LINKED ISSUE #44096: [Epic] Elevate “include this period” in time filters
**Links**
- [Product doc](https://www.notion.so/metabase/Elevate-include-this-period-in-time-filter-b828dcca50b7491c9f1719d2bf6c60a0?pvs=4)
- Eng doc - TDB
- Feature branch:  `qc-proj-elevate-include-this-period`
- [Figma](https://www.figma.com/design/ofKULr1X7TQf5MZCkbJZvx/Elevate-%E2%80%9Cinclude-this-period%E2%80%9D-in-time-filter?node-id=1-286&t=vXLIPw5F50pJoQyr-0)
- Testing plan: #44323

# Implementation Plan
This should be fairly self-contained epic that most likely doesn't need more than one milestone. It has three distinct places in the UI that need to be updated. Maybe it makes sense to merge them all together to `master`, or maybe they are distinct enough which allows merging them individually. I'll know this when I dig into the code.

## Milestone 1
Update and re-work the existing frontend/src/metabase/querying/components/DatePicker/RelativeDatePicker/DateIntervalPicker/DateIntervalPicker.tsx component.
```[tasklist]
- [x] Add the new option to the filter in the notebook mode (#44179)
- [x] Add the new option to the filter modal in the simple mode (#44179)
- [x] Add the new option to the filter in the drill-through context menu (simple mode) (#44179)
- [x] Elevate the "Starting from " icon/action (#44242)
- [x] Merge milestone 1 to `master` (#44279)
```

## Milestone 2
Time-series chrome component.
Implementing this will resolve #42220

```[tasklist]
- [x] Add the new option to the time-series chrome in the charts/data visualizations (#44247)
- [x] Cover new changes with more tests (#44247)
```

## Milestone 3
Milestone 1 and Milestone 2 follow-ups.
> [!Note]
> I propose we merge the Milestone 1 and 2 to `master` to give product and design teams a few days to feel the new changes before we start polishing them.
```[tasklist]
- [x] Update copy (Past -> Previous) ([Slack context](https://metaboat.slack.com/archives/C02H619CJ8K/p1718358798489699)) (#44299)
- [ ] Tweak the design (#44554)
```

### Additional Notes
The related issue/feature request: https://github.com/metabase/metabase/issues/42220

### PR DESCRIPTION
**Pull request by nemanjaglumac**
### Description

This PR changes how the TimeSeriesChrome component look and feel like in subtle, yet impactful ways.
1. It stacks the operator picker on top of the actual filter pickers
2. It reuses the same CurrentDate picker logic that we're using in other date pickers in the app

The component was very wide in the past for the Interval date pickers, such as "Previous" or "Next" because the operator picker was rendered in a row next to the actual filter picker.

This PR completes the Milestone 3 of #44096, and it resolves #44096.

### How to verify
Tests already exist for all these interactions so they should pass in CI.

If you want to manually play with this, follow these steps:
1. Open Orders table and summarize it by Count, broken down by Created At: month
2. Time-series chrome component should be visible in the lower part of the screen

From here you can choose either
- "Previous" or "Next", or
- Current

### Demo
**Interval Picker**

Before
image

After
image

**Current Picker**

Before
image

After
image

### Checklist

- [x] Tests have been added/updated to cover changes in this PR

### Additional Notes
Figma conversation that sparked this: https://www.figma.com/design/ofKULr1X7TQf5MZCkbJZvx?node-id=4030-3370#801341512

### CODE REVIEW

File: frontend/src/metabase/querying/components/DatePicker/DateOperatorPicker/DateOperatorPicker.tsx [hunk 1]
```diff
@@ -1,14 +1,13 @@
 import { useMemo } from "react";
 
-import { Group, Stack } from "metabase/ui";
+import { Stack, Select } from "metabase/ui";
```
Comment (kamilmielnik) [inline frontend/src/metabase/querying/components/DatePicker/DateOperatorPicker/DateOperatorPicker.tsx:3]: The [padding around the pills](https://github.com/metabase/metabase/assets/6830683/b3d74681-71a3-440d-b6ec-3e2ad34f6dda) looks a bit out of place, IMO it [looks better without it](https://github.com/metabase/metabase/assets/6830683/42139ad2-63c5-4a4d-a71c-6dfe63a6cbf5).

Comment (nemanjaglumac) [reply frontend/src/metabase/querying/components/DatePicker/DateOperatorPicker/DateOperatorPicker.tsx:3]: Spot on!
Thank you <3

Comment (AlessioLaiso) [reply frontend/src/metabase/querying/components/DatePicker/DateOperatorPicker/DateOperatorPicker.tsx:3]: Agreed! Thanks!

Comment (nemanjaglumac) [reply frontend/src/metabase/querying/components/DatePicker/DateOperatorPicker/DateOperatorPicker.tsx:3]: This was just another consequence of the different implementations between `RelativeDatePicker` vs `SimpleRelativeDatePicker`.

The former lets its children deal with the padding, while the latter has a padding on the parent/container.

Comment (nemanjaglumac) [reply frontend/src/metabase/querying/components/DatePicker/DateOperatorPicker/DateOperatorPicker.tsx:3]: Fixed in https://github.com/metabase/metabase/pull/44554/commits/1d4876bb2261d34922c33206619653c454ad5cc3

image

image

**Review by AlessioLaiso — APPROVED** [review]
Agreed with the comment on padding around the pills. Looks good overall!

File: frontend/src/metabase/querying/components/DatePicker/RelativeDatePicker/CurrentDatePicker/CurrentDatePicker.tsx [hunk 2]
```diff
@@ -26,7 +31,7 @@ export function CurrentDatePicker({ value, onChange }: CurrentDatePickerProps) {
   };
 
   return (
-    <Stack p="md">
+    <Stack p={padded ? "md" : undefined}>
```
Comment (ranquild) [inline frontend/src/metabase/querying/components/DatePicker/RelativeDatePicker/CurrentDatePicker/CurrentDatePicker.tsx:34]: Please remove `padded` from props and add padding where it's needed and where the component is used. Maybe with a `Box`

**Review by ranquild — APPROVED** [review]
One comment about `padded`

File: frontend/src/metabase/querying/components/DatePicker/RelativeDatePicker/CurrentDatePicker/CurrentDatePicker.tsx (hunk 2 above)
Comment (ranquild) [reply frontend/src/metabase/querying/components/DatePicker/RelativeDatePicker/CurrentDatePicker/CurrentDatePicker.tsx]: lgtm

**Pull request merged**